File: sc823.Custom/Commands/EditPageDetails.cs
using Score.Custom.Commands;

namespace sc823.Custom.Commands
{
    public class ManageCustomData : ScoreFieldEditorCommand
    {
        public ManageCustomData()
        {
            var articleId = "{734B09D5-5412-4173-B1CC-BE6295BB30D7}";

            Dialog = new FieldEditorConfiguration
            {
                ScopeToTemplates = new string[] { articleId },
                FieldsSectionsToEdit = new string[] { "Page Details" },
                Title = "Assign stuff to the current page",
                Width = "750",
                Height = "900"
            };
        }
    }
}

File: sc823.Web/sitecore/shell/client/Sitecore/ExperienceEditor/Commands/EditPageDetails.js
require.config({
    paths: {
        scoreFieldEditor: "/-/speak/v1/score/FieldEditor"
    }
});

define(["sitecore", "scoreFieldEditor"], function (Sitecore, Score) {
    Sitecore.Commands.EditPageDetails = {
        canExecute: function (context) {
            if (context.currentContext.webEditMode !== "edit") {
                return false;
            }

            context.currentContext.argument = "{734B09D5-5412-4173-B1CC-BE6295BB30D7}"; // Custom Page Data

            return context.app.canExecute("ExperienceEditor.Score.IsScorePage", context.currentContext);
        },

        execute: function (context) {
            context.app.disableButtonClickEvents();

            Score.FieldEditor({
                saveItem: false,
                header: "Manage Custom Data",
                title: "Assign Custom Data to the current page",
                sections: "Page Details",
                width: "900px",
                height: "750px"
            }).execute(context);

            context.app.enableButtonClickEvents();
        }
    };
});